openhcl/hcl_mapper/src/lib.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Provides a mapper implementation for the page pool that uses [`MshvVtlLow`].
+
+#![cfg(target_os = "linux")]
+#![warn(missing_docs)]
+
+use anyhow::Context;
+use hcl::ioctl::MshvVtlLow;
+use hvdef::HV_PAGE_SIZE;
+use page_pool_alloc::Mapper;
+use page_pool_alloc::PoolType;
+use sparse_mmap::SparseMapping;
+
+/// A mapper that uses [`MshvVtlLow`] to map pages.
+pub struct HclMapper {
+    fd: MshvVtlLow,
+}
+
+impl HclMapper {
+    /// Creates a new [`HclMapper`].
+    pub fn new() -> Result<Self, anyhow::Error> {
```

Comment:
I mean, I guess anyhow::Error isn't too bad... but you could use the concrete type as well

Reply:
I think the extra context here is useful, because it seems weird to me at least to return that type. I know we don't do anything else today, but it's not really expected for the caller to handle this error other than bubble it up.